retrace/glretrace.py

```diff
@@ -196,6 +198,24 @@ def retraceFunctionBody(self, function):
 
         Retracer.retraceFunctionBody(self, function)
 
+        # If we have a query buffer or if we execute the query we have to loop until the
+        # query result is actually available to get the proper result - for the following
+        # execution if the query buffer is used or for the check to make sense, and if we
+        # just want to execute the query for timing purpouses we also should wait
+        # for the result.
+        if function.name.startswith('glGetQueryObject'):
+           print(r'    if (_query_buffer || retrace::queryHandling != retrace::QUERY_SKIP) {')
```

Comment:
I think this should be `!_query_buffer` no?

When there's a query buffer bound, the value is not written to `params`, but rather written into the query buffer, into the offset given by `params`.  See https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/glGetQueryObject.xhtml

Reply:
In fact, I think this shoud be 

```c++
    if (!_query_buffer && retrace::queryHandling != retrace::QUERY_SKIP) {
```

Reply:
I know, I just had a laps in logic, not thinking  that for checking whether the result available it is follows the same rule. But there was also an error when the query result call was run but the result not supposed to be checked. I have updated the MR accordingly.